Week4/Solution.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Alyssa Lingad
+ * 
+ * MICP: Week 4
+ */
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import java.util.*;
+
+
+
+public class Solution
+{
+	public static boolean canBeSegmented(String str, HashSet d) throws Exception{
+		int i = 0;
+		if (str == null)
+			throw new Exception();
+		if (str.isEmpty())
+			return false;
+		if (d.isEmpty())
+			return false;
+		
+		int size = str.length();
+		String s = str.toLowerCase();
+		boolean check = false;
+		while (i < size){
+			int j = i + 2;
+			check = false; 
+			while (j < size+1){
+				//System.out.println(s.substring(i, j));
+				if (d.contains(s.substring(i, j))){
+					check = true;
+					break;
+				}
+				j++;
+			}
+			if (!check)
+				return false;
+			if (j == size)
+				return true;
+			i = j;
+		}
+		return check;
+	}
+	
```

Comment:
your code didn't work for this example:

"a"
["a"]

Can you make sure if it works for all other test cases or not?